build.rs

```diff
@@ -42,4 +42,8 @@ fn main() {
             "The `print-jtag-serial` feature is only supported by the ESP32-C3, ESP32-C6, ESP32-S3 and ESP32-H2 chips"
         );
     }
+
+    if cfg!(feature = "semihosting") && cfg!(feature = "halt-cores") {
```

Comment:
Could we also bail out if we enable semihosting on an xtensa target?

Reply:
The `semihosting` crate already bails out if you try to use it on a target which is not supported.
https://github.com/taiki-e/semihosting/blob/main/src/lib.rs#L223C1-L233

Would you like an additional check here anyways?